app/js/lib/foxbox.js

```diff
@@ -173,6 +219,66 @@ export default class Foxbox extends Service {
       // Throwing here to abort the promise chain.
       throw(new Error('Redirecting to a URL without session'));
     }
+
+    return Promise.resolve();
+  }
+
+  /**
+   * Attach event listeners related to the connection status.
+   *
+   * @return {Promise}
+   * @private
+   */
+  _initNetwork() {
+    window.addEventListener('online', this.pingBox.bind(this));
+    window.addEventListener('offline', this.pingBox.bind(this));
+
+    if ('connection' in navigator && 'onchange' in navigator.connection) {
+      navigator.connection.addEventListener('change', this.pingBox.bind(this));
+
+      // We also ping the box every few minutes to make sure it's still there.
+      _onlineInterval = setInterval(this.pingBox.bind(this),
```

Comment:
To detect if the box gets disconnected from the internet (unplugged, turned off, power outage...).